_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp

```diff
@@ -5104,6 +5111,73 @@ mfxU64 make_def_back_color_yuv(mfxU16 bit_depth)
         ;
 };
 
+mfxU64 GetBackgroundColor(const mfxVideoParam &videoParam)
+{
+    mfxU64 iBackgroundColor;
+    mfxU64 def_back_color = 0xffff000000000000;
+
+    if (videoParam.vpp.Out.FourCC == MFX_FOURCC_NV12 ||
+        videoParam.vpp.Out.FourCC == MFX_FOURCC_YV12 ||
+        videoParam.vpp.Out.FourCC == MFX_FOURCC_NV16 ||
+        videoParam.vpp.Out.FourCC == MFX_FOURCC_YUY2 ||
+        videoParam.vpp.Out.FourCC == MFX_FOURCC_AYUV )
+    {
+        def_back_color = make_def_back_color_yuv(8);
+    }
+    else if(videoParam.vpp.Out.FourCC == MFX_FOURCC_P010 ||
+            #if (MFX_VERSION >= 1027)
+            videoParam.vpp.Out.FourCC == MFX_FOURCC_Y210 ||
+            videoParam.vpp.Out.FourCC == MFX_FOURCC_Y410 ||
+            #endif
+            videoParam.vpp.Out.FourCC == MFX_FOURCC_P210)
+    {
+        def_back_color = make_def_back_color_yuv(10);
+    }
+
+    iBackgroundColor = def_back_color;
```

Comment:
I don't understand why you need new iBackgroundColor. You can continue to use def_back_color.